BUILD.gn

```diff
@@ -63,6 +69,21 @@ rust_test("handlers_test") {
     "$rust_build:libc",
     "$rust_build:url",
     "$rust_build:log",
+
+    # Indirect rust depdendencies also need to be listed here:
+    # * Linking to `:handlers` or `:libdeno` isn't possible, because they
+    #   already contain some symbols exported by `handlers.rs`. These duplicate
+    #   symbols trip up the linker.
+    # * The `rust_test` and `rust_executable` templates only produce an object
+    #   file, and then invoke an external linker. Transitive rust depencenies
+    #   are not resolved in either step.
+    "$rust_build:idna",
+    "$rust_build:percent_encoding",
+    "$rust_build:unicode_bidi",
+    "$rust_build:unicode_normalization",
```

Comment:
This is unfortunate. What if we moved all the tests into their own file - `handlers_test.rs` ?

Reply:
IDK - I thought of that too but I didn't try. I suspect its gonna be very brittle.

Another meh option is to generate a list of all crates in rust/BUILD.gn, and just pass that to all targets.

Maybe best to let it stew for a while and then ask some rust person for their advice.

Reply:
> best to let it stew for a while 
agree
> Another meh option is to generate a list of all crates in rust/BUILD.gn
I believe fuchsia does something like this.